c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -680,12 +826,21 @@ class DeviceCachingAllocator {
       }
     }
 
+    // Dirty but simple check if the new cudaMalloc is for a CUDA graph's PrivatePool
+    for (auto& gp : graph_pools) {
+      if (p.pool == &gp.second.large_blocks ||
+          p.pool == &gp.second.small_blocks) {
+        gp.second.cudaMalloc_count++;
+        break;
+      }
```

Comment:
This is going to be O(n) cost for number of live cuda graphs, even if you aren't recording, no?

Reply:
Yes, but we only got here if we called cudaMalloc, which dwarfs everything else in this function.  This function is already the slow (hopefully rare) fallback.

Reply:
new `BlockPool.owner_PrivatePool` member simplified this function.